core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java

```diff
@@ -472,19 +481,20 @@ public static List<String> getMetadataObjectLocation(
                     .catalogDispatcher()
                     .loadCatalog(
                         NameIdentifier.of(ident.namespace().level(0), ident.namespace().level(1)));
-            LOG.info("Catalog provider is %s", catalogObj.provider());
-            if (catalogObj.provider().equals("hive")) {
+            LOG.info("Catalog provider is {}", catalogObj.provider());
+            String catalogProvider = catalogObj.provider();
+            if (Objects.equals(catalogProvider, "hive")
+                || Objects.equals(catalogProvider, "hadoop")) {
```

Comment:
Why don't we just do string comparison?

Reply:
It's easy to get NPE for a new starter like stringVariable.equals("value") if `stringVariable` is a null value.